src/runtime/StateSerialization/RuntimeData.cs

```diff
@@ -82,32 +84,37 @@ static void ClearCLRData ()
 
         internal static void Stash()
         {
-            var runtimeStorage = new PythonNetState
-            {
-                Metatype = MetaType.SaveRuntimeData(),
-                ImportHookState = ImportHook.SaveRuntimeData(),
-                Types = TypeManager.SaveRuntimeData(),
-                Classes = ClassManager.SaveRuntimeData(),
-                SharedObjects = SaveRuntimeDataObjects(),
-            };
-
-            IFormatter formatter = CreateFormatter();
-            var ms = new MemoryStream();
-            formatter.Serialize(ms, runtimeStorage);
-
-            Debug.Assert(ms.Length <= int.MaxValue);
-            byte[] data = ms.GetBuffer();
-            // TODO: use buffer api instead
-            IntPtr mem = PyMem_Malloc(ms.Length + IntPtr.Size);
-            Marshal.WriteIntPtr(mem, (IntPtr)ms.Length);
-            Marshal.Copy(data, 0, mem + IntPtr.Size, (int)ms.Length);
-
             ClearCLRData();
 
-            using NewReference capsule = PyCapsule_New(mem, IntPtr.Zero, IntPtr.Zero);
-            int res = PySys_SetObject("clr_data", capsule.BorrowOrThrow());
-            PythonException.ThrowIfIsNotZero(res);
-            PostStashHook?.Invoke();
+            IFormatter? formatter = CreateFormatter();
+
+            if (formatter != null)
```

Comment:
Can't we just exit the method if formatter is null? Would avoid polluting git history

Reply:
Sure, will do.